pkg/controller/core/admissioncheck_controller.go

```diff
@@ -196,14 +209,16 @@ func (h *acCqHandler) Generic(ctx context.Context, e event.GenericEvent, q workq
 	log := log.FromContext(ctx).WithValues("clusterQueue", klog.KObj(cq))
 	log.V(6).Info("Cluster queue generic event")
 
-	for _, ac := range cq.Spec.AdmissionChecks {
-		if cqs := h.cache.ClusterQueuesUsingAdmissionCheck(ac); len(cqs) == 0 {
-			req := reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Name: string(ac),
-				},
+	if cq.Spec.AdmissionChecksStrategy != nil {
```

Comment:
Omission of handling `cq.Spec.AdmissionChecksStrategy` here looks like a bug prior to the PR. I think it would be better to submit as a preparatory PR, or just cherrypick this bit. Let me know if I'm missing something.